packages/connection-form/src/utils/csfle-handler.ts

```diff
@@ -152,29 +178,25 @@ export function handleUpdateCsfleKmsTlsParam({
   const autoEncryption = connectionOptions.fleOptions?.autoEncryption ?? {};
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const tls: any = {
-    ...(autoEncryption.tlsOptions?.[action.kmsProvider] ?? {}),
+    ...(autoEncryption.tlsOptions?.[action.kmsProviderName] ?? {}),
   };
   if (!action.value) {
     delete tls[action.key];
   } else {
     tls[action.key] = action.value;
   }
   const tlsOptions = autoEncryption.tlsOptions ?? {};
-  if (Object.keys(tls).length === 0) {
-    delete tlsOptions[action.kmsProvider];
-  } else {
-    tlsOptions[action.kmsProvider] = tls;
-  }
+  tlsOptions[action.kmsProviderName] = tls;
```

Comment:
Our of curiousity, what's bringing on this change? If the KMS provider name changes, does this "leak" TLS options that do not correspond to any of the KMS providers?

Reply:
ah, this i missed completely when renaming. i'll ensure in rename, tlsoption is also renamed. 
and this should remain as original code. will revert this change.

Reply:
reverted in 1b179f689066e3d4948304a2345e0fad9335c7fa